files/scripts/syncd.sh

```diff
@@ -72,15 +72,21 @@ start() {
 
         # Flush DB during non-warm start
         /usr/bin/docker exec database redis-cli -n 1 FLUSHDB
+    fi
+
+    # platform specific tasks
+
+    if [ x$sonic_asic_platform == x'mellanox' ]; then
+        export FAST_BOOT=1
+        /usr/bin/mst start
+        /usr/bin/mlnx-fw-upgrade.sh
+        /etc/init.d/sxdkernel start
+        /sbin/modprobe i2c-dev
+    fi
+
 
-        # platform specific tasks
-        if [ x$sonic_asic_platform == x'mellanox' ]; then
-            export FAST_BOOT=1
-            /usr/bin/mst start
-            /usr/bin/mlnx-fw-upgrade.sh
-            /etc/init.d/sxdkernel start
-            /sbin/modprobe i2c-dev
-        elif [ x$sonic_asic_platform == x'cavium' ]; then
+    if [[ x"$WARM_BOOT" != x"true" ]]; then
+        if [ x$sonic_asic_platform == x'cavium' ]; then
```

Comment:
could you not move code for other platform?

Reply:
This diff looks very confusion. But it seems that mellanox changed one if ! warm boot block into 3 blocks.

It appears that mellanox want to do firmware upgrade and other tasks regardless if it is ffb or not. Can you confirm this is intentional?

If it is intentional, can you leave a comment to state reason?

Reply:
If you plan to fix something for cavium, please contact them first.

Reply:
I don't understand the concern, since I did not change the behavior for other platform.
Previously it was under else block of if warm boot block. Now I changed it to if ! warm boot.

The reason I split into two blocks is to split platform specific section and platform-independent section.

Starting sxdkernel, mlnx-fw-upgrade, etc. is intentional. Above line "export FAST_BOOT=1" indicates that ASIC will not be reset during driver start.
In case of cold boot ASIC is reset during create_switch.
I added a comment that this is intentional